rllib/tests/backward_compat/test_backward_compat.py

```diff
@@ -0,0 +1,50 @@
+import unittest
+
+
+class TestBackwardCompatibility(unittest.TestCase):
+    def test_register_all(self):
+        """Tests the old (1.10) way of registering all Trainers.
+
+        Uses the old 1.10 registry.py file and thus makes sure all Trainers can still
+        be imported using their old paths (albeit this will create a warning).
+        """
+        # Try importing old Trainer class (this is just an Alias now to the `Algorithm`
+        # class).
+        from ray.rllib.agents.trainer import Trainer  # noqa
+
+        # Old registry code.
+        from ray.rllib.tests.backward_compat.old_registry import (
+            ALGORITHMS,
+            _get_trainer_class,
+        )
+        from ray.rllib.contrib.registry import CONTRIBUTED_ALGORITHMS
+
+        # Test the old `_get_trainer_class()` utility that was used to pull Trainer
+        # class and default config.
+        for key in (
+            list(ALGORITHMS.keys())
+            + list(CONTRIBUTED_ALGORITHMS.keys())
+            + ["__fake", "__sigmoid_fake_data", "__parameter_tuning"]
+        ):
+            _get_trainer_class(key)
+
+        # Also test this for all `CONTRIBUTED_ALGORITHMS`.
+        for key in list(CONTRIBUTED_ALGORITHMS.keys()):
+            _get_trainer_class(key)
+
+    def test_old_configs(self):
```

Comment:
Don't you wanna do this for all algos in the combined list of `ALGORITHMS` and `CONTRIBUTED_ALGORITHMS` from agents / contrib? why only ppo?

Reply:
I think this would blow up the test significantly if we tested ALL algos in here. The main point is to test the generic (shared) Trainer settings with one example algo.